### PR TITLE
[AIP-192] Rule to forbid raw HTML.

### PR DESCRIPTION
**Pull request by lukesneeringer**

### CODE REVIEW

File: docs/rules/0192/no-html.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+# No HTML in comments
+
+This rule enforces that every descriptor in every proto file does not use raw
+HTML in comments, as mandated in [AIP-192][].
+
+## Details
+
+This rule looks at each descriptor in each proto file (exempting oneofs and the
+file itself) and tries to pick up "HTML smell", and complains if it thinks it
```
Comment (mingzhi) [inline docs/rules/0192/no-html.md]: Consider to rephrase it to: "tries to find HTML tags, and complains if it found some." Basically, we need to explain what is "HTML smell".

File: docs/rules/0192/no-html.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+found HTML.
+
+**Note:** This lint rule uses a regular expression to look for HTML, which is a
```
Comment (mingzhi) [inline docs/rules/0192/no-html.md]: Consider to rephrase it to "**Note:** May miss some complex HTML. ..."

File: docs/rules/0192/no-html.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  string name = 1;
+}
+```
+
+**Correct** code for this rule:
+
+```proto
+// Correct.
+// A representation of a book.
+// A representation of a book.
```
Comment (mingzhi) [inline docs/rules/0192/no-html.md]: Should we remove this duplicated line?

File: rules/aip0192/no_html.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	"regexp"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var noHTML = &lint.DescriptorRule{
+	Name: lint.NewRuleName(192, "no-html"),
+	LintDescriptor: func(d desc.Descriptor) []lint.Problem {
+		for _, comment := range separateInternalComments(d.GetSourceInfo().GetLeadingComments()).External {
```
Comment (mingzhi) [inline rules/aip0192/no_html.go]: Add a `OnlyIf` `d.GetSourceInfo() != nil`.

File: rules/aip0192/no_html.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+// Yes, yes, I know: https://stackoverflow.com/questions/1732348/
+// Even Jon Skeet cannot parse HTML using regular expressions.
+//
+// That said, we really only want to pick up "basic HTML smell", and are
+// disinterested in actually doing any manipulation, and we can be at least
+// a little tolerant of false positives/negatives.
+//
+// Therefore, in this case, a regex seems better than taking a dependency
+// just for this.
+var htmlSmell = regexp.MustCompile(`</?[a-zA-Z]+( /)?>`)
```
Comment (mingzhi) [inline rules/aip0192/no_html.go]: Consider to rename `htmlSmell` to more informative, like `htmlTagRegexp`.

File: docs/rules/0192/no-html.md (hunk 1 above)
Comment (lukesneeringer) [reply docs/rules/0192/no-html.md]: Done.

File: docs/rules/0192/no-html.md (hunk 2 above)
Comment (lukesneeringer) [reply docs/rules/0192/no-html.md]: _Stet._

File: docs/rules/0192/no-html.md (hunk 3 above)
Comment (lukesneeringer) [reply docs/rules/0192/no-html.md]: Yep, thanks. Done.

File: rules/aip0192/no_html.go (hunk 4 above)
Comment (lukesneeringer) [reply rules/aip0192/no_html.go]: Good idea. That also requires adding `OnlyIf` to `DescriptorRule`. Done.

File: rules/aip0192/no_html.go (hunk 5 above)
Comment (lukesneeringer) [reply rules/aip0192/no_html.go]: Done.
